cmd/kpod/common.go

```diff
@@ -50,30 +78,59 @@ func getStore(c *cli.Context) (storage.Store, error) {
 	return store, nil
 }
 
+func getCopyOptions(reportWriter io.Writer, signaturePolicyPath string, srcDockerRegistry, destDockerRegistry *dockerRegistryOptions, signing signingOptions) *cp.Options {
+	if srcDockerRegistry == nil {
+		srcDockerRegistry = &dockerRegistryOptions{}
+	}
+	if destDockerRegistry == nil {
+		destDockerRegistry = &dockerRegistryOptions{}
+	}
+	srcContext := srcDockerRegistry.getSystemContext(signaturePolicyPath)
+	destContext := destDockerRegistry.getSystemContext(signaturePolicyPath)
+	return &cp.Options{
+		RemoveSignatures: signing.RemoveSignatures,
+		SignBy:           signing.SignBy,
+		ReportWriter:     reportWriter,
+		SourceCtx:        srcContext,
+		DestinationCtx:   destContext,
+	}
+}
+
+func getPolicyContext() (*signature.PolicyContext, error) {
+	var systemContext types.SystemContext
+	policy, err := signature.DefaultPolicy(&systemContext)
```

Comment:
`copy.Options.{Source,Destination}Ctx` apply to accessing the source/destination transports, where `SignaturePolicyPath` does not matter and is ignored (this is admittedly unclear, and a disadvantage of having a single `SystemContext` for everything).

For `copy.Image`, only the explicit `policyContext` argument is used to select the policy. It has to be created with `NewPolicyContext` using a `*signature.Policy`. Now, to get the `*Policy`, the following options exist:

- `signature.DefaultPolicy(nil)`: let the system choose
- `signature.NewPolicyFromFile(path)`: explicitly load a file
- `signature.DefaultPolicy(ctx)`, with `ctx.SignaturePolicyPath` empty or pointing to a file
- create it manually or mostly-manually in code.

The third option, i.e. `signature.DefaultPolicy(&types.SystemContext{SignaturePolicyPath:optionalPath})`, or `ctx := getSystemContext(); /*other users of ctx*/; signature.DefaultPolicy(ctx)` is probably the most convenient one to integrate with the CLI flag handling.